agent-inject/metrics.go

```diff
@@ -0,0 +1,54 @@
+package agent_inject
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+	metricsNamespace      = "vault"
+	metricsSubsystem      = "sidecar_injector"
+	metricsLabelNamespace = "namespace"
+)
+
+var (
+	requestQueue = prometheus.NewGauge(prometheus.GaugeOpts{
+		Namespace: metricsNamespace,
+		Subsystem: metricsSubsystem,
+		Name:      "request_queue_total",
+		Help:      "Total count of webhook requests in the queue",
+	})
+
+	requestProcessingTime = prometheus.NewHistogram(prometheus.HistogramOpts{
+		Namespace: metricsNamespace,
+		Subsystem: metricsSubsystem,
+		Name:      "request_processing_duration_ms",
+		Help:      "Histogram of webhook request processing times in milliseconds",
+		Buckets: []float64{
```

Comment:
Went with a fixed set of buckets, similar to [what's done for VSO](https://github.com/hashicorp/vault-secrets-operator/blob/1273434f78c5a2561ec461715306d146cc13fe69/vault/client_metrics.go#L24), but we could also use something like [ExponentialBuckets](https://pkg.go.dev/github.com/prometheus/client_golang/prometheus#ExponentialBuckets)?

Reply:
Decided to stick with a predefined set of buckets but updated the configuration to reduce their number (reducing cardinality) and increase the highest-tier bucket to 10 seconds (from 1 second).